lib/pure/times.nim

```diff
@@ -2553,6 +2583,28 @@ proc `+=`*(t: var Time, b: TimeInterval) =
 proc `-=`*(t: var Time, b: TimeInterval) =
   t = t - b
 
+#
+# Day of year
+#
+
+proc initDateTime*(weekday: WeekDay, yearweek: YearweekRange, year: int,
+                   hour: HourRange, minute: MinuteRange, second: SecondRange,
+                   nanosecond: NanosecondRange,
+                   zone: Timezone = local()): DateTime =
+  ## Create a new `DateTime <#DateTime>`_ from a year, week number, and weekday
+  ## in the specified timezone.
+
+  # source https://webspace.science.uu.nl/~gent0113/calendar/isocalendar.htm
+  let d = yearweek * 7 + weekday.int - initDateTime(4, mJan, year, 00, 00, 00).weekday.int - 4
+  initDateTime(1, mJan, year, hour, minute, second, nanosecond, zone) + initTimeInterval(days=d)
+
+template initDateTime*(weekday: WeekDay, yearweek: YearweekRange, year: int,
+                   hour: HourRange, minute: MinuteRange, second: SecondRange,
+                   zone: Timezone = local()): DateTime =
```

Comment:
why do we need this overload? just use `nanosecond: NanosecondRange = 0` above

Reply:
We already have two other `initDateTime` procs anyway, one of which is unnecessary, I assume @capocasa copied that from there?

Reply:
I did it that way for consistency with the existing API.

The alias proc isn't really unnecessary, it allows you to create a datetime by positional arguments with a time zone without specifying nanoseconds. I assume the original author concluded time zones were a more common use case than nanoseconds, and I agree. But even if you don't, having to remember which kind of initDateTime allows omitting nanoseconds with a time zone would suck, so the alias should stay for consistency's sake, expectations are there, too late to change them.

Reply:
we shouldn't aggravate the problem, let's just introduce 1 overload, not 2
```nim
proc initDateTime*(weekday: WeekDay, yearweek: YearweekRange, year: int,
                   hour: HourRange, minute: MinuteRange, second: SecondRange,
                   nanosecond: NanosecondRange,
                   zone: Timezone = local()): DateTime {.since: (1, 5).} =
```

in future work, we can discuss whether to deprecate the pre-existing
```nim
proc initDateTime(monthday: MonthdayRange; month: Month; year: int;
                  hour: HourRange; minute: MinuteRange; second: SecondRange;
                  zone: Timezone = local())
```
and make `nanoseconds` an option param for
```nim
proc initDateTime(monthday: MonthdayRange; month: Month; year: int;
                  hour: HourRange; minute: MinuteRange; second: SecondRange;
                  nanosecond: NanosecondRange; zone: Timezone = local())
```